Remove id_tax_rules_group in carrier table

jf-viguier opened the pull request:
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | id_tax_rules_group is not used anymore.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     |  Fixes https://github.com/PrestaShop/PrestaShop/issues/25208
| How to test?      | remove column in database without this pr and test an order and stock changes. 
| Possible impacts? | none

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/25207)
<!-- Reviewable:end -->

jf-viguier commented:
Is this column can be deleted or it should be a bc break ?

Progi1984 commented:
@jf-viguier Sorry... But I have another parts in code linked to carriers :
* `admin-dev/themes/default/template/controllers/carriers/helpers/form/form.tpl` (l.37)
* `controllers/admin/AdminCarriersController.php` a lot of `id_tax_rules_group`
* `js/admin/carrier_wizard.js` l.197
* `UI/pages/BO/shipping/carriers/add.js` l.26

jf-viguier commented:
@Progi1984 id_tax_rules_group is stored for the carrier in another table than carrier table : 
 carrier_tax_rules_group_shop.id_tax_rules_group

jf-viguier commented:
@Progi1984 thanks to automatic tests, I've removed id_tax_rules_group from initial data

jf-viguier commented:
@Progi1984 all tests are ok now, please remove Waiting for author label as all id_tax_rules_group you founded are related to the other table, not carrier table.

jf-viguier committed:
Remove id_tax_rules_group in carrier table

jf-viguier commented:
what's the problem with Integration tests (7.4) ?

Progi1984 commented:
@jf-viguier Restarted CI.

PierreRambaud reviewed (changes requested):
Upgrade scripts need to be updated :)

jf-viguier commented:
@PierreRambaud nice catch, added in 1.7.9 upgrade

PierreRambaud commented on install-dev/upgrade/sql/1.7.9.0.sql:
```diff
@@ -9,3 +9,4 @@ INSERT INTO `PREFIX_configuration` (`name`, `value`, `date_add`, `date_upd`) VAL
 ;
 INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES
   (NULL, 'actionValidateOrderAfter', 'New Order', 'This hook is called after validating an order by core', '1');
+ALTER TABLE `PREFIX_carrier` DROP COLUMN `id_tax_rules_group`;
```
```suggestion
ALTER TABLE `PREFIX_carrier` DROP COLUMN `id_tax_rules_group`;

```

jf-viguier commented:
Thanks @PierreRambaud

jolelievre reviewed (dismissed):
@jf-viguier there is still a reference to this DB field here
https://github.com/PrestaShop/PrestaShop/blob/7c005444cf00710a81149e6c5ca861f32c600ce6/classes/Carrier.php#L606

Even if the function doesn't seem to be used by the core, but it might be used by modules So either we remove it (which is a BC break) or we can update the query to rely on the other table `carrier_tax_rules_group_shop`

jf-viguier commented:
@jolelievre actually, id_tax_rules_group is always 0 in carrier table, so this method always return false.

jf-viguier committed:
Fix getIdTaxRulesGroupMostUsed

jf-viguier commented:
@jolelievre I've fixed the getIdTaxRulesGroupMostUsed sql. It's used in AdminCarrierWizardController.php.

jf-viguier commented:
It's seems that I've fixed some CRLF in Carrier.php

matthieu-rolland reviewed (commented):
seems like there are conflicts to be solved, otherwise LGTM 👍

jf-viguier commented:
@matthieu-rolland conflicts resolved

jf-viguier commented:
@Progi1984 @matthieu-rolland what's wrong with tests ?

matks commented:
> @Progi1984 @matthieu-rolland what's wrong with tests ?

prestashop.com is down so tests are in another dimension 😅

Julievrz commented:
Looks like there's no wording in this PR, I'm removing the "waiting for wording" label. Feel free to ping me if needed. 🤗

matks commented:
Hi @jf-viguier sorry for coming back late. Unfortunately in September prestashop.com was down so it messed up the CI. I close and reopen the PR to reset the CI and know whether tests pass however you'll need to rebase the PR.

PierreRambaud commented on classes/Carrier.php:
```diff
@@ -599,15 +599,15 @@ public static function getIdTaxRulesGroupMostUsed()
+				ORDER BY n DESC
+				LIMIT 1
+			) most_used'
         );
```
Be careful, you replace all spaces with tabs for no reason.

jf-viguier replied:
@PierreRambaud it's fixed.

Progi1984 reviewed (dismissed):
@jf-viguier Could you move the SQL query to `autoupgrade` module ?

PierreRambaud commented:
Hi @jf-viguier ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

jf-viguier commented:
@PierreRambaud what are you waiting for ? I've done the pr you asked ?
https://github.com/jf-viguier/autoupgrade/pull/1

jf-viguier commented:
@PierreRambaud my bad, the pr is there https://github.com/PrestaShop/autoupgrade/pull/437 (wrong depot)

jf-viguier commented:
@Progi1984 asked pr is done.

PierreRambaud commented:
I reopen this PR, but next time, don't forget to answer maintainers' messages and link to the right repository :wink:

PierreRambaud commented:
Anyway the PR still need to be rebased

matks reviewed (approved):
Looks OK but we need to be extra careful about validating this change. PrestaShop is not friendly with such changes; we might break something in an unknown location.

@jf-viguier can you please provide a "How to test" a little bit more complete for QA team ? it could mention places in the software where id_tax_rules_group might be used. It must be for them a path they can follow to look for issues.

jf-viguier commented:
@matks this column is not used anymore since 1.7

matks commented:
@jf-viguier Do you mean there is, according to our knowledge, zero line of code that do use this column in prestashop ?

jf-viguier commented:
@matks yes this column is not used anymore since 1.7, The only place where it's used is in this stat method Carrier::getIdTaxRulesGroupMostUsed(). I've fixed it n the pr: this stat always return zero.
I've updated the "how to test".

khouloudbelguith commented:
Hi @jf-viguier,

It is ok :heavy_check_mark: 
1. I created some carriers with different ranges
2. I created a carrier with a tax breakdown
3. I created some orders from the FO & BO 
4. I created a cart rule "free shipping" & apply it
5. I edited some orders when "Recalculate shipping costs after editing the order" is enabled & disabled

=> ok :heavy_check_mark: 
https://watch.screencastify.com/v/7T2BNIjZYutavq0jQklP

**With PR**
![image](https://user-images.githubusercontent.com/16067358/144880542-048ab8d7-5805-4344-8e31-9aac7304db37.png)

**Without PR**
![image](https://user-images.githubusercontent.com/16067358/144880614-052eb4fd-c229-488c-a5aa-a19f5150ea5f.png)

Thanks!

Progi1984 commented:
Thanks @jf-viguier & @khouloudbelguith

jf-viguier commented:
Thanks @Progi1984, don't forget this related pr here to delete the column https://github.com/PrestaShop/autoupgrade/pull/437
